commit fc996db970a33c74d3db3ee63532b15187258027
Merge: 60573c2966a1 2e6cfd496f5b
Author: Linus Torvalds <torvalds@linux-foundation.org>
Date:   Thu Oct 22 13:00:44 2020 -0700

    Merge tag 'vfio-v5.10-rc1' of git://github.com/awilliam/linux-vfio
    
    Pull VFIO updates from Alex Williamson:
    
     - New fsl-mc vfio bus driver supporting userspace drivers of objects
       within NXP's DPAA2 architecture (Diana Craciun)
    
     - Support for exposing zPCI information on s390 (Matthew Rosato)
    
     - Fixes for "detached" VFs on s390 (Matthew Rosato)
    
     - Fixes for pin-pages and dma-rw accesses (Yan Zhao)
    
     - Cleanups and optimize vconfig regen (Zenghui Yu)
    
     - Fix duplicate irq-bypass token registration (Alex Williamson)
    
    * tag 'vfio-v5.10-rc1' of git://github.com/awilliam/linux-vfio: (30 commits)
      vfio iommu type1: Fix memory leak in vfio_iommu_type1_pin_pages
      vfio/pci: Clear token on bypass registration failure
      vfio/fsl-mc: fix the return of the uninitialized variable ret
      vfio/fsl-mc: Fix the dead code in vfio_fsl_mc_set_irq_trigger
      vfio/fsl-mc: Fixed vfio-fsl-mc driver compilation on 32 bit
      MAINTAINERS: Add entry for s390 vfio-pci
      vfio-pci/zdev: Add zPCI capabilities to VFIO_DEVICE_GET_INFO
      vfio/fsl-mc: Add support for device reset
      vfio/fsl-mc: Add read/write support for fsl-mc devices
      vfio/fsl-mc: trigger an interrupt via eventfd
      vfio/fsl-mc: Add irq infrastructure for fsl-mc devices
      vfio/fsl-mc: Added lock support in preparation for interrupt handling
      vfio/fsl-mc: Allow userspace to MMAP fsl-mc device MMIO regions
      vfio/fsl-mc: Implement VFIO_DEVICE_GET_REGION_INFO ioctl call
      vfio/fsl-mc: Implement VFIO_DEVICE_GET_INFO ioctl
      vfio/fsl-mc: Scan DPRC objects on vfio-fsl-mc driver bind
      vfio: Introduce capability definitions for VFIO_DEVICE_GET_INFO
      s390/pci: track whether util_str is valid in the zpci_dev
      s390/pci: stash version in the zpci_dev
      vfio/fsl-mc: Add VFIO framework skeleton for fsl-mc devices
      ...

diff --cc arch/s390/pci/pci_bus.c
index 0c0db7c3a404,c93486a9989b..755b46f4c595
--- a/arch/s390/pci/pci_bus.c
+++ b/arch/s390/pci/pci_bus.c
@@@ -135,9 -197,10 +135,10 @@@ void pcibios_bus_add_device(struct pci_
  	 * With pdev->no_vf_scan the common PCI probing code does not
  	 * perform PF/VF linking.
  	 */
- 	if (zdev->vfn)
+ 	if (zdev->vfn) {
 -		zpci_bus_setup_virtfn(zdev->zbus, pdev, zdev->vfn);
 +		zpci_iov_setup_virtfn(zdev->zbus, pdev, zdev->vfn);
- 
+ 		pdev->no_command_memory = 1;
+ 	}
  }
  
  static int zpci_bus_add_device(struct zpci_bus *zbus, struct zpci_dev *zdev)